### ISSUE TITLE
Incorrect typings for ReadableStream

### ISSUE THREAD

**Issue by ondras** [denoland/deno#7217]
Hi,

my code uses the following tsconfig `lib` declaration:
```
"lib": ["dom", "dom.iterable", "deno.ns"],
```

after updating to Deno 1.3.1, the following TS error emerges:

```
error: TS2339 [ERROR]: Property 'getIterator' does not exist on type 'ReadableStream<R>'.
  return res.readable.getIterator();
                      ~~~~~~~~~~~
    at https://deno.land/std@0.66.0/async/pool.ts:45:23
```

...which is preventing me from running the code.

Thanks,
O.

**Comment by lucacasonato** [comment]
I don't think this is a Deno issue. It seems the 'dom' type library (from typescript) doesn't include types for getIterator. So this looks like a upstream typescript issue.

**Comment by kitsonk** [comment]
Actually, we have a challenge.  The [specification was updated in June](https://github.com/whatwg/streams/commit/caf3cea28134d8be2ec69b6a1105748a7c8d10b5) and `getIterator()` was renamed to `values()`.  It was never implemented in Chrome as `getIterator()`.  I guess this is the problem with implementing perpetually draft specifications. 🤷

**Comment by kitsonk** [comment]
Actually looking at that now, there is a whole load of changes there... 😭  it needs to be gone through with a fine toothed comb and realigned.

**Comment by ondras** [comment]
Is there at least a way to temporarily monkey-patch this (such as `interface ReadableStream { getIterator ... }`) so the code using the *dom* lib can be executed?

**Comment by kitsonk** [comment]
Interfaces are open ended.  Something like this should work in your main module, assuming it is TypeScript:

```ts
declare global {
  interface ReadableStream<R = any> {
    getIterator(options?: { preventCancel?: boolean }): AsyncIterableIterator<R>;
  }
}
```

**Comment by ondras** [comment]
> Interfaces are open ended. Something like this should work in your main module, assuming it is TypeScript:

Great, thanks @kitsonk!

(my naïve attempts were missing the `declare global` statement, thus failing to fix stuff)

**Comment by lucacasonato** [comment]
> Actually, we have a challenge.  The [specification was updated in June](https://github.com/whatwg/streams/commit/caf3cea28134d8be2ec69b6a1105748a7c8d10b5) and `getIterator()` was renamed to `values()`.  It was never implemented in Chrome as `getIterator()`.  I guess this is the problem with implementing perpetually draft specifications. 🤷 
> 
> 

Oh no! I was worried this was the case. This is going to definitely be a breaking change to fix, and it is going to break std types again 😭.

**Comment by kitsonk** [comment]
I suspect we can have `values()` and `getIterator()` for now and move all `std` over to `values()`.

**Comment by timonson** [comment]
This error still comes up. Is there a workaround?

**Comment by ondras** [comment]
> This error still comes up. Is there a workaround?

The code mentioned in https://github.com/denoland/deno/issues/7217#issuecomment-682370049 is a suitable workaround for my use case.

**Comment by timonson** [comment]
It works indeed, thank you @ondras ! I misread kitsonk's comment the first time.

**Comment by lucacasonato** [comment]
This was resolved in #9103.
